backend/backend.go

```diff
@@ -689,6 +693,15 @@ func (backend *Backend) Banners() *banners.Banners {
 // param can be any string, as it is potentially passed without any validation from the calling
 // platform.
 func (backend *Backend) HandleURI(uri string) {
-	// TODO: handle
-	backend.log.Debugf("Handle URI: %s", uri)
+	u, err := url.Parse(uri)
+	if err != nil {
+		backend.log.WithError(err).Warningf("Handling URI failed: %s", uri)
+		return
+	}
+	switch u.Scheme {
+	case "aopp":
+		backend.handleAOPP(u)
```

Comment:
I would pass a copy here:

```go
backend.handleAOPP(*u)
```

avoid worring about null pointer dereference in handleAOPP. https://play.golang.org/p/d9YQrWZJeYb

Reply:
On the other hand it copies lots of bytes for no reason?

Reply:
Done